packages/ui-causal-graph-editor/src/shared/rendering/utils.tsx

```diff
@@ -79,3 +79,20 @@ export function getZoomState(scale: number, zoomThresholds?: ZoomThresholds): Zo
 export function isGraphLayoutWithTiers(layout: GraphLayout): layout is GraphLayoutWithTiers {
     return (layout as GraphLayoutWithTiers).tiers !== undefined;
 }
+
+export function isGraphLayoutWithGroups(layout: GraphLayout): layout is GraphLayoutWithGrouping {
+    return (layout as GraphLayoutWithGrouping).group !== undefined;
+}
+
+export function findKeyByValue(obj: Record<string, string[]>, searchValue: string): string {
+    if (Object.values(obj).flat().includes(searchValue)) {
+        let returnKey: string;
+        Object.keys(obj).forEach((key) => {
+            if (obj[key].includes(searchValue)) {
+                returnKey = key;
+            }
+        });
+        return returnKey;
+    }
+    return searchValue;
+}
```

Comment:
That's a fairly inefficient implementation of the search, as you're effectively searching for it twice. Can you think of a more efficient implementation? Try to reduce the number of iterations over the lists/objects, and make sure to stop the search whenever an item is found.